### ISSUE TITLE
Actual fullheight for heros

### ISSUE THREAD

**Issue by dergoegge** [jgthms/bulma#1949]
This is about **Bulma**.

It's a bug concerning the ```is-fullheight``` class.

It's not a browser bug.
### Overview of the problem

This is about the Bulma **CSS framework** 

 I'm using Bulma **version** [0.7.1]
 My **browser** is: Chrome, Opera

 I am sure this issue is **not a duplicate**

### Description

A fullheight hero is not fullheight if the navbar is fixed to the top.

### Steps to Reproduce

1. Add ```has-navbar-fixed-top``` to the class list of the html attribute.
2. Set up a page as follows: 

```html
<section class='hero is-fullheight'>
    <div class="hero-body">
	  <div class="container">
	      <h1 class='title'>Title</h1>
	  </div>
      </div>
   <div class='hero-footer'>
	<div class="container">
	     <p>Not on the screen unless scrolled to</p>
	</div>
   </div>
</section >
```

### Expected behavior

I expected to see the hero-footer on screen.

### Actual behavior

The hero-footer will only be visible if scrolled to.

**Comment by jimblue** [comment]
Hi @dergoegge,

Her is my hack for this issue:

```scss
.hero.is-fullheight navbar-is-fixed {
    min-height: calc(100vh - #{$navbar-height});
}
```

**Comment by dergoegge** [comment]
Not sure if this is actually a bug or intended behaviour...

**Comment by jgthms** [comment]
This could be added as a modifier class.

**Comment by jimblue** [comment]
@jgthms that could be nice :)

**Comment by emkajeej** [comment]
wouldn't it be easier to use the body/html class, instead of adding another helper to the hero?
ie

```css
.has-navbar-fixed-top .hero.is-fullheight {
    min-height: calc(100vh - #{$navbar-height});
}
```

**Comment by jimblue** [comment]
@emkajeej I think a modifier is be better here as people there's probably particularly cases where  people would prefer to have the choice.

**Comment by emkajeej** [comment]
I'm inclined to disagree. The docs state that a `.is-fullheight` modifier makes the hero as large as the viewport, so it's only fair that the user expects a component with the exact height of the viewport, not "the height plus a seemingly arbitrary  amount of pixels".
Secondly, the `.has-fixed-navbar` naming implies that the navbar is part of the hero, while in reailty it's part of the `body`.

In the edge case that the user actually wants the `.is-fullheight` hero to be slightly larger than the actual viewport the `.has-navbar-fixed-top .hero.is-fullheight` is easily overridden.

**Comment by frederikhors** [comment]
Can we have this (https://github.com/jgthms/bulma/issues/1949#issuecomment-399421960) code merged, @jgthms?

**Comment by jgthms** [comment]
> The docs state that a .is-fullheight modifier makes the hero as large as the viewport, so it's only fair that the user expects a component with the exact height of the viewport, not "the height plus a seemingly arbitrary amount of pixels".

That's not true. The fullheight hero is basically never higher than the viewport. The height is set with `min-height: 100vh`. I'm using `min-height` and not `height` because I don't want the content to be clipped if it's higher than the viewport. But that's almost never the case, and not the case here.

Example: https://codepen.io/jgthms/pen/ReRPpe

The hero is fullheight here. You can't see the footer because the `has-navbar-fixed-top` class on the `<html>` adds padding at the top. But you can see that the hero _is_ fullheight (I made the navbar transparent). So:

* `has-navbar-fixed-top` is doing what's it's supposed to
* `is-fullheight` is doing what's it's supposed to

It would be disingenuous to make the `is-fullheight` class make the hero _less_ than fullheight.

The solution is with an additional modifier for the hero, something like the one stated here: https://github.com/jgthms/bulma/issues/1949#issuecomment-398594152

**Comment by dergoegge** [comment]
@jgthms The PR #1954 i opened does not change the ```is-fullheight``` class. It adds another modifier to the hero like you proposed early on. Maybe take a look and merge or give me a few pointers as to what more needs to be done so we can get this issue resolved!

**Comment by jgthms** [comment]
Thanks @dergoegge that's pretty good actually! Will have a proper look later.

**Comment by emkajeej** [comment]
> The solution is with an additional modifier for the hero, something like the one stated here: [#1949 (comment)](https://github.com/jgthms/bulma/issues/1949#issuecomment-398594152)

In the above solution the hero height is also being modified to be less than the viewport.
i'm proposing exactly that, but with the modifier being on the body rather than the hero for the reasons stated in the second half of my comment (ie semantically and for consistency).

**Comment by claide** [comment]
Encountered a weird behavior using `is-fullheight-with-navbar` to the hero element.
Everything works but when adding the same hero to another page, the hero element is residing behind the navbar fixed top.

**Comment by i-defranca** [comment]
> Encountered a weird behavior using `is-fullheight-with-navbar` to the hero element.
> Everything works but when adding the same hero to another page, the hero element is residing behind the navbar fixed top.

I'm having this problem too. Did you find a solution?

**Comment by i-defranca** [comment]
> Encountered a weird behavior using `is-fullheight-with-navbar` to the hero element.
> Everything works but when adding the same hero to another page, the hero element is residing behind the navbar fixed top.

@claide I'm having this problem too. Did you find a solution?

**Comment by claide** [comment]
> > Encountered a weird behavior using `is-fullheight-with-navbar` to the hero element.
> > Everything works but when adding the same hero to another page, the hero element is residing behind the navbar fixed top.
> 
> @claide I'm having this problem too. Did you find a solution?

Hmm I forgot about this, but try to add class `has-navbar-fixed-top` to your html tag like below

`<html class="has-navbar-fixed-top">`

**Comment by ompande** [comment]
> This is about **Bulma**.
> 
> It's a bug concerning the `is-fullheight` class.
> 
> It's not a browser bug.
> 
> ### Overview of the problem
> This is about the Bulma **CSS framework**
> 
> I'm using Bulma **version** [0.7.1]
> My **browser** is: Chrome, Opera
> 
> I am sure this issue is **not a duplicate**
> 
> ### Description
> A fullheight hero is not fullheight if the navbar is fixed to the top.
> 
> ### Steps to Reproduce
> 1. Add `has-navbar-fixed-top` to the class list of the html attribute.
> 2. Set up a page as follows:
> 
> ```
> <section class='hero is-fullheight'>
>     <div class="hero-body">
> 	  <div class="container">
> 	      <h1 class='title'>Title</h1>
> 	  </div>
>       </div>
>    <div class='hero-footer'>
> 	<div class="container">
> 	     <p>Not on the screen unless scrolled to</p>
> 	</div>
>    </div>
> </section >
> ```
> 
> ### Expected behavior
> I expected to see the hero-footer on screen.
> 
> ### Actual behavior
> The hero-footer will only be visible if scrolled to.

Thanks! It worked.
